src/CMakeLists.txt

```
@@ -1,6 +1,32 @@
-# Copyright (c) 2023 The Bitcoin Core developers
+# Copyright (c) 2023-present The Bitcoin Core developers
 # Distributed under the MIT software license, see the accompanying
-# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+# file COPYING or https://opensource.org/license/mit/.
+
+#=============================
+# secp256k1 subtree
+#=============================
+set(SECP256K1_DISABLE_SHARED ON CACHE BOOL "" FORCE)
+set(SECP256K1_ENABLE_MODULE_ECDH OFF CACHE BOOL "" FORCE)
+set(SECP256K1_ENABLE_MODULE_RECOVERY ON CACHE BOOL "" FORCE)
+set(SECP256K1_ECMULT_GEN_KB 86 CACHE BOOL "" FORCE)
```

Is there anything else we can/should be turning off here. i.e if we never intend to compile or run any tests, then we don't need to check for valgrind, or similar.

---

Autotools currently disables the benchmarks.
```suggestion
set(SECP256K1_BUILD_BENCHMARK OFF CACHE BOOL "" FORCE)
```

The tests are currently enabled. Are these run as part of any test or CI in Core? (I don't think so.) So we could also disable them.  `SECP256K1_BUILD_TESTS`, `SECP256K1_BUILD_EXHAUSTIVE_TESTS`, `SECP256K1_BUILD_CTIME_TESTS` 

(I may be a good idea to run the tests occasionally, e.g., when creating official builds. But that's an orthogonal issue.)

---

> we don't need to check for valgrind

I've added setting `SECP256K1_VALGRIND`.

Speaking of `SECP256K1_BUILD_*` variables, `EXCLUDE_FROM_ALL` does this job and @theuni's [opinion](https://github.com/hebasto/bitcoin/pull/192#discussion_r1594428809) was:
> We could remove the `SECP256K1_BUILD_*` options and not have to worry about any new binaries being added in the future.

---

> > we don't need to check for valgrind
> 
> I've added setting `SECP256K1_VALGRIND`.
> 

Ah, sorry, I should have commented on this instead of just ignoring it.  Can you change it back?

This has a history: It is one specific instance of the discussion in https://github.com/bitcoin-core/secp256k1/issues/1549, where our decision (a few years ago) was to leave Valgrind enabled in production builds (if possible), to make sure that the `ctime_tests` binary with Valgrind on matches production builds as closely as possible, see https://github.com/bitcoin-core/secp256k1/pull/813#issuecomment-691720230 and the comment below it for detailed rationale.

---

> Speaking of `SECP256K1_BUILD_*` variables, `EXCLUDE_FROM_ALL` does this job and @theuni's [opinion](https://github.com/hebasto/bitcoin/pull/192#discussion_r1594428809) was:
> 
> > We could remove the `SECP256K1_BUILD_*` options and not have to worry about any new binaries being added in the future.

Oh sure, I forgot!

---

> to make sure that the ctime_tests binary with Valgrind on matches production builds as closely as possible, see https://github.com/bitcoin-core/secp256k1/pull/813#issuecomment-691720230 and the comment below it for detailed rationale.

Does this still apply if we aren't (currently) actually building or running any tests (it's not super clear to me if this is the right decision)?

---

Yes, I think so. The library is written in a way such that the constant-time tests have a hidden switch to tell the library to do some Valgrind magic. This was really done with the idea that you can run the constant-time tests against the actual production library (you can literally link it to shared library) without any code changes (e.g., alignment, size, etc.) that could influence the compiler's decision to optimize or not optimize our constant-time hacks.  

So under the assumption that the same "misoptimization" happens with both the compiler used in production and the compiler used in libsecp256k1 testing, this is useful. (But sure, all of this would be even better if we ran the constant-time tests here because then the compilers would be the same and the assumption would hold trivially. Perhaps we should open an issue for this. Building and running the ctime_tests binary takes less than a second on my machine, so this should not be a burden for the test suite in Core.)

Does this make sense?

---

> (But sure, all of this would be even better if we ran the constant-time tests here because then the compilers would be the same and the assumption would hold trivially. Perhaps we should open an issue for this.

---

> (But sure, all of this would be even better if we ran the constant-time tests here because then the compilers would be the same and the assumption would hold trivially. Perhaps we should open an issue for this.

I guess this is fairly related to https://github.com/bitcoin-core/secp256k1/issues/1549, and actually running tests against code, when varying from the default secp256k1 configs, so maybe we can continue discussion there? I was planning no commenting soon, and think we should have some sort of agreement on what we are doing before moving ahead here without running any tests at all.

---

So I think `SECP256K1_VALGRIND` should be left on (=status quo in autotools) for now, at least until we come to a different conclusion in   https://github.com/bitcoin-core/secp256k1/issues/1549 .

---

@real-or-random 
> So I think `SECP256K1_VALGRIND` should be left on (=status quo in autotools) for now, at least until we come to a different conclusion in [bitcoin-core/secp256k1#1549](https://github.com/bitcoin-core/secp256k1/issues/1549) .

`set(SECP256K1_VALGRIND ON)` will require valgrind installed, which is not always the case, at least in the Guix environment.

I suggest to skip setting `SECP256K1_VALGRIND` value from here.

---

> I suggest to skip setting `SECP256K1_VALGRIND` value from here.

Ok yes, that's what I meant, sorry.

---

> > I suggest to skip setting `SECP256K1_VALGRIND` value from here.
> 
> Ok yes, that's what I meant, sorry.

Thanks! [Done](https://github.com/hebasto/bitcoin/pull/192#issuecomment-2196678927).